### ISSUE TITLE
Review LSST zeropoints

### ISSUE THREAD

**Issue by dkirkby** [DarkEnergyScienceCollaboration/WeakLensingDeblending#19]
Calculate values for the latest LSST filters with airmass 1.2 using:
```
from speclite.filters import load_filter, ab_reference_flux

def calculate_zero_point(band_name, B0=24):
    filt = load_filter(band_name)
    return (filt.convolve_with_function(ab_reference_flux) *
            10 ** (-0.4 * B0)).to(1 / (u.s * u.m**2))
```
This gives 1.349 / (s m2) compared with 1.681 currently in the code (which is based on airmass 1).  Can the airmass difference can explain this, or did the filter curves change that much?  The original values were committed in Dec 2014.

**Comment by fjaviersanchez** [comment]
The filter's throughput several times during 2015 (I had to revert to the commit `2aa3ef709f841b85eb10eeeff13c1526c841e8ee` in `https://github.com/lsst/throughputs`) that makes the difference between 1.681 and 1.349. The new filter response curves include updates in the engineering model and the change from airmass 1.0 to 1.2 so, it looks like a combination of an updated model and an updated airmass.

![filter_response](https://cloud.githubusercontent.com/assets/13971732/21742735/ab60a6c8-d4f4-11e6-9674-bfe6c53b3d3b.png)

**Comment by dkirkby** [comment]
If its easy, can you make the same plot with both filters at airmass 1 by dividing out the airmass-only contribution (assuming there is a separate file for that)?  It would also be useful to compare the airmass-only curve with the simple extinction model assumed in the code (which will need to be updated if the nominal airmass has changed).

**Comment by fjaviersanchez** [comment]
I computed several things:

1) The ratio of the throughput from the updated atmosphere model with aerosol and airmass =1.2 divided by the throughput with airmass = 1.0 and no aerosols

![ratio_x0](https://cloud.githubusercontent.com/assets/13971732/21744978/3de38cfa-d521-11e6-805d-5434acc2ef66.png)

2) A plot showing the new filter with airmass 1.2 and the old filter multiplied by the correction from the plot above. They match better but there is still something else updated.

![ratio_filters](https://cloud.githubusercontent.com/assets/13971732/21744986/72ec3c62-d521-11e6-8928-a26e20ca03d1.png)

I also updated the zeropoint values in the paper draft. Should I also update anything on the program?

**Comment by fjaviersanchez** [comment]
I computed the flux with the correction `ab_magnitude +=extinction*(airmass-1.0)` and without this correction but computing a new zeropoint. To do so, I used the airmass throughputs from `lsst/throughputs` for r-band at X=1.4 the difference in flux counts is ~1.5%
![dif_mag](https://cloud.githubusercontent.com/assets/13971732/21745774/518c8ae2-d533-11e6-9bf5-69ceb3bc800f.png)

**Comment by dkirkby** [comment]
It looks like the change of airmass + aerosols only accounts for a ~5% change, but there is an additional ~10% change from other sources.  Can you track this down in the files that specify the other contributions to the overall throughput: detector, filter_r, hardware_r, lens1-3 and m1-3?

When you wrote X=1.4 did you mean X=1.2?  I only see values of 1.0 and 1.2 tabulated in `lsst/throughputs`.

**Comment by fjaviersanchez** [comment]
Yes, I will. I meant X=1.4 you can see the values tabulated from 1.0 to 2.5 in [here](https://github.com/lsst/throughputs/tree/master/atmos)

**Comment by fjaviersanchez** [comment]
It looks like one of the mirrors was missing in the previous version of the throughputs (the one that we used to calculate the zeropoints in the package originally). See the plots attached:

![throughputs](https://cloud.githubusercontent.com/assets/13971732/21763265/4fd03d0e-d65d-11e6-8535-3cf1c77c8424.png)

![components](https://cloud.githubusercontent.com/assets/13971732/21763316/a0de3250-d65d-11e6-8175-df5860134577.png)

**Comment by fjaviersanchez** [comment]
Solved in #20

**Comment by sowmyakth** [comment]
I used the LSST default values currently in descwl/survey.py on the catsim galaxies and I got the magnitude off by 0.05. I think the cause is the zero point value. The LSST r band zero point used here is 1.349 / (s m2). The effective area mentioned in the code is 32.4 (m2), which gives zero point = 43.7.  However the value of r band zero point used in the code is 44.8. I think the reason is that in #20 the zero point was updated to 44.8 on Jan 9. The effective area was changed from 33.212 to 32.4 on Jan 13, but the zero point wasn't updated.  Is this correct or have I missed something in the code?
